AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/TerrainSystem_VegetationSpawnsOnTerrainSurfaces.py

```diff
@@ -0,0 +1,217 @@
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+class VegetationTests:
+    vegetation_on_gradient_1 = (
+        "Vegetation detected at correct position on Gradient1", 
+        "Vegetation not detected at correct position on Gradient1"
+    )
+    vegetation_on_gradient_2 = (
+        "Vegetation detected at correct position on Gradient2", 
+        "Vegetation not detected at correct position on Gradient2"
+    )
+    unfiltered_vegetation_count_correct = (
+        "Unfiltered vegetation spawn count correct", 
+        "Unfiltered vegetation spawn count incorrect"
+    )
+
+    testTag2_excluded_vegetation_count_correct = (
+        "TestTag2 filtered vegetation count correct", 
+        "TestTag2 filtered vegetation count incorrect"
+    )
+    testTag2_excluded_vegetation_z_correct = (
+        "TestTag2 filtered vegetation spawned in correct position", 
+        "TestTag2 filtered vegetation failed to spawn in correct position"
+    )
+
+    testTag3_excluded_vegetation_count_correct = (
+        "TestTag3 filtered vegetation count correct", 
+        "TestTag3 filtered vegetation count incorrect"
+    )
+    testTag3_excluded_vegetation_z_correct = (
+        "TestTag3 filtered vegetation spawned in correct position", 
+        "TestTag3 filtered vegetation failed to spawn in correct position"
+    )
+
+    cleared_excludion_vegetation_count_correct = (
+        "Cleared filter vegetation count correct", 
+        "Cleared filter vegetation count incorrect"
+    )
+
+def TerrainSystem_VegetationSpawnsOnTerrainSurfaces():
+    """
+    Summary:
+    Load an empty level, 
+    Create two entities with constant gradient components with different values.
+    Create two entities with TerrainLayerSpawners
+    Create an entity to spawn vegetation
+    Ensure that vegetation spawns at the correct heights
+    Add a VegetationSurfaceMaskFilter and ensure it responds correctly to surface changes.
+    :return: None
+    """
+
+    import os
+    import sys
+    import math as sys_math
+
+    import azlmbr.legacy.general as general
+    import azlmbr.bus as bus
+    import azlmbr.math as math
+
+    import azlmbr.areasystem as areasystem
+    import azlmbr.editor as editor
+    import azlmbr.vegetation as vegetation
+    import azlmbr.terrain as terrain
+    import azlmbr.entity as EntityId
+    import azlmbr.surface_data as surface_data
+
+    import editor_python_test_tools.hydra_editor_utils as hydra
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+
+    def create_entity_at(entity_name, components_to_add, x, y, z):
+        entity = hydra.Entity(entity_name)
+        entity.create_entity(math.Vector3(x, y, z), components_to_add)
+
+        return entity
+
+    def FindHighestAndLowestZValuesInArea(aabb):
+        vegetation_items = areasystem.AreaSystemRequestBus(bus.Broadcast, 'GetInstancesInAabb', aabb)
+
+        lowest_z = sys.float_info.max
```

Comment:
I think you could save a few lines here by doing something like

lowest_z = min([item.position.z for item in vegetation_items])

Reply:
Done